packages/components/src/List/Toolbar/SelectSortBy/SelectSortBy.component.js

```diff
@@ -75,16 +77,31 @@ function SelectSortBy({ field, id, isDescending, onChange, options, t }) {
 				</NavDropdown>
 			)}
 			{selected && (
-				<NavItem
-					id={id && `${id}-order`}
-					onClick={onChangeOrder}
-					aria-label={t('LIST_CHANGE_SORT_BY_ORDER', {
-						defaultValue: 'Change sort order. Current order: {{sortOrder}}.',
-						sortOrder: currentSortOrderLabel,
-					})}
-				>
-					{currentSortOrderLabel}
-				</NavItem>
+				<li>
+					<Button
+						id={id ? `${id}-order-chooser` : uuid.v4()}
```

Comment:
Don't generate one if no props.id is not provided.
```suggestion
						id={id && `${id}-order-chooser`}
```